src/Essentials/src/MediaPicker/MediaPicker.uwp.cs

```diff
@@ -67,5 +77,80 @@ public async Task<FileResult> CaptureAsync(MediaPickerOptions options, bool phot
 
 			return null;
 		}
+
+		class WinUICameraCaptureUI
+		{
+			const string WindowsCameraAppPackageName = "Microsoft.WindowsCamera_8wekyb3d8bbwe";
+
+			public WinUICameraCaptureUIPhotoCaptureSettings PhotoSettings { get; } = new();
+
+			public WinUICameraCaptureUIVideoCaptureSettings VideoSettings { get; } = new();
+
+			public async Task<StorageFile> CaptureFileAsync(CameraCaptureUIMode mode)
+			{
+				var hwnd = WindowStateManager.Default.GetActiveWindowHandle(true);
+
+				var options = new LauncherOptions();
+				InitializeWithWindow.Initialize(options, hwnd);
+
+				options.TreatAsUntrusted = false;
+				options.DisplayApplicationPicker = false;
+				options.TargetApplicationPackageFamilyName = WindowsCameraAppPackageName;
+
+				var extension = mode == CameraCaptureUIMode.Photo
+					? PhotoSettings.GetFormatExtension()
+					: VideoSettings.GetFormatExtension();
+
+				var tempLocation = await StorageFolder.GetFolderFromPathAsync(FileSystem.CacheDirectory);
+				var tempFile = await tempLocation.CreateFileAsync($"capture{extension}", CreationCollisionOption.GenerateUniqueName);
+				var token = global::Windows.ApplicationModel.DataTransfer.SharedStorageAccessManager.AddFile(tempFile);
+
+				var set = new ValueSet();
+				if (mode == CameraCaptureUIMode.Photo)
+				{
+					set.Add("MediaType", "photo");
+					set.Add("PhotoFileToken", token);
+				}
+				else
+				{
+					set.Add("MediaType", "video");
+					set.Add("VideoFileToken", token);
+				}
+
+				var uri = new Uri("microsoft.windows.camera.picker:");
+				var result = await WinLauncher.LaunchUriForResultsAsync(uri, options, set);
+				if (result.Status == LaunchUriStatus.Success && result.Result != null)
+					return await StorageFile.GetFileFromPathAsync(tempFile.Path);
```

Comment:
This call is throwing with:

> onecore\shell\shcore\libs\stream\randomaccessstream.cpp(1518)\shcore.dll!00007FFF2BAEE223: (caller: 00007FFF2BACC936) LogHr(4) tid(27770) 80070323 The handle with which this oplock was associated has been closed.  The oplock is now broken.

Reply:
I think this may have been because my cache location is my destination so I am opening a file twice and reading and writing to it simultaneously...

Reply:
yup, was that.